### PR TITLE
Apiml basic routing

### PR DESCRIPTION
**Pull request by taban03**
#### Is there a related issue for this PR? 
Issue number:

### CODE REVIEW

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+**Examples:**
+
+* `z/OSMF direct URL: `https://ca32.ca.com:1443/zosmf/restjobs/1.0/...`
+* `Gateway URL: `https://ca3x.ca.com:10010/api/zosmfca32/restjobs/1.0/...`
+
+The following diagram presents how basic routing works:
+<img src="../../images/api-mediation/Basic-Routing.png" alt="Zowe API Mediation basic routing"/> 
+
+
```
Comment (janan07) [inline docs/extend/extend-apiml/api-mediation-routing.md:49]: There should be further description of how the services are routed as shown in this diagram.

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+* The service ID is used to find the service host and port. 
+* The service-url is used to prefix the resourcePath at the service host.
+
+## Basic Routing (only service ID-based)
+
+This method of routing is the same as the previous method but does not use the version part of the URL. This is useful for services that handle their versioning themselves with different granularity.
+An example of this is z/OSMF.
+
+**Example?**
+
```
Comment (janan07) [inline docs/extend/extend-apiml/api-mediation-routing.md:89]: This seems incomplete. There should be an example of this type of basic routing.

File: docs/extend/extend-apiml/api-mediation-routing.md (hunk 1 above)
Comment (taban03) [reply docs/extend/extend-apiml/api-mediation-routing.md:49]: Hmm not sure what else I can add to be honest, I don't think it should go too deep in detail.

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# API Mediation Layer routing
+
+As an application developer, route your service to ...
+There are two ways to route to the API Mediation Layer:
```
Comment (balhar-jakub) [inline docs/extend/extend-apiml/api-mediation-routing.md:4]: Isn't something missing here?

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+* API ML Basic Routing (Service ID and version-based)
+* Basic Routing (only service ID-based)
+
+## Terminology
+
+* **Service**
+
+  A service provides one or more APIs. A service is identified by its service ID (sometimes the term service name is used in the same     meaning).
```
Comment (balhar-jakub) [inline docs/extend/extend-apiml/api-mediation-routing.md:13]: Please remove unnecessary spces

File: docs/extend/extend-apiml/api-mediation-routing.md (hunk 3 above)
Comment (taban03) [reply docs/extend/extend-apiml/api-mediation-routing.md:4]: Yeah, it was added by Andrew as he is trying to find a good introduction sentence

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Basic routing is based on the service ID that is used to identify the service. The specific instance is selected by the API Gateway. Identical response are required for all instances. Eureka and Zuul expect this type of routing.
+
+The URI identifies the resource. It does not identify the instance of the same service as unique when multiple instances of the same service are provided, such as when the service is running in HA mode. 
+Services of the same product that are providing different resources, such as SYSVIEW on one system and SYSVIEW in a different sysplex, cannot have the same service ID as the same URI cannot have two different meanings. 
+
+In addition to the basic Zuul routing, the Zowe API Gateway adds support for versioning where a major version can be specified. 
+The Gateway routes a request only to an instance that provides the specified major version of the API. 
+The `/api/` prefix is used for REST APIs. The prefix `/ui/` prefix applies to web UIs and the prefix `/ws/` applies to WebSockets.
+
+Additional routing can be implemented using a Zuul Pre-filter.
```
Comment (balhar-jakub) [inline docs/extend/extend-apiml/api-mediation-routing.md:33]: Remove this

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+```
+http://service:10015/enablerv1sampleapp/api/v1/samples
+```
+API Gateway URL of the resource
+```
+https://gateway:10010/api/v1/enablerv1sampleapp/samples
+```
+
+The following diagram presents how basic routing works:
+<img src="../../images/api-mediation/Basic-Routing.png" alt="Zowe API Mediation basic routing"/>
```
Comment (balhar-jakub) [inline docs/extend/extend-apiml/api-mediation-routing.md:51]: The diagram isn't shown here in PR, is the path correct?

File: docs/extend/extend-apiml/api-mediation-routing.md (hunk 5 above)
Comment (taban03) [reply docs/extend/extend-apiml/api-mediation-routing.md]: Why, is it not valid?

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+**Note:** The service ID is not included in the routing metadata, but the service ID is in the basic Eureka metadata. 
+
+## Basic Routing (using only the service ID)
+
+This method of routing is the same as the previous method but does not use the version part of the URL. This is useful for services that handle their versioning themselves with different granularity.
+An example of this is z/OSMF.
+
+**Example:**
+
+z/OSMF URL through the Gateway: `https://gateway:10010/api/zosmf/restjobs/1.0/...`
```
Comment (janan07) [inline docs/extend/extend-apiml/api-mediation-routing.md:94]: Can you please add a description of what is contained in this address?

File: docs/extend/extend-apiml/api-mediation-routing.md (hunk 4 above)
Comment (taban03) [reply docs/extend/extend-apiml/api-mediation-routing.md]: It is actually intentional as it is part of the Service bullet point

File: docs/extend/extend-apiml/api-mediation-routing.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+In this URL example, `zosmf` is the service ID, and `/restjobs/1.0/...` is the rest of the endpoint segment.
+No version is specified in this URL.
```
Comment (janan07) [inline docs/extend/extend-apiml/api-mediation-routing.md:97]: If the 1.0 in this example does not refer to a version number, shouldn't we explain what this does refer to?

Comment (taban03) [reply docs/extend/extend-apiml/api-mediation-routing.md:97]: No it's not related to the API Version, but I agree that it is confusing, and actually this specific z/OSMF endpoint should be /zosmf1/zosmf/restjobs/jobs. It was just an example anyway, but good catch, I can modify it to `https://gateway:10010/api/zosmf/restjobs/jobs/...`

**Pull request merged**